### PR TITLE
OpenAPI Definition for thirdparty-api-adapter

### PR DESCRIPTION
**Pull request by sridharvoruganti**
`#316 : Added following API definitions , please review it.`

**New endpoints**

1. consentRequest (**3**) : `POST /consentRequests, PUT /consentRequests/{ID} and PUT /consentRequests/{ID}/error`
2. consent (**4**):  `POST /consents, GET /consents/{ID} ,PUT /consents/{ID}  and POST /consents/{ID}/createCredential`
3. thirdPartyRequests (**4**): `POST /thirdPartyRequests/transactions, GET /thirdPartyRequests/transactions/{ID}, PUT /thirdPartyRequests/transactions/{ID} and PUT /thirdPartyRequest/transactions/{ID}/error`

**Existing endpoints from Mojaloop API**

1. authorizations (**3**) : `POST /authorizations, PUT /authorizations/{ID}  and PUT /authorizations/{ID}/error`
2. parties (**6**): `GET /parties/{Type}/{ID}, PUT /parties/{Type}/{ID}, PUT /parties/{Type}/{ID}/error, GET /parties/{Type}/{ID}/{SubId},  PUT /parties/{Type}/{ID}/{SubId} and PUT /parties/{Type}/{ID}/{SubId}/error`

### CODE REVIEW

File: src/interface/thirdparty-api.yaml [hunk 1]
```diff
@@ -0,0 +1,2352 @@
...
+          $ref: '#/components/responses/404'
+        405:
+          $ref: '#/components/responses/405'
+        406:
+          $ref: '#/components/responses/406'
+        501:
+          $ref: '#/components/responses/501'
+        503:
+          $ref: '#/components/responses/503'
+  /metrics:
```
Comment (lewisdaly) [inline src/interface/thirdparty-api.yaml:46]: I suppose this is boilerplate, but we should leave it in as we may want it in the future.

**Comment by lewisdaly** [comment]
Thanks @sridharvoruganti. This is looking good enough for now - I'm guessing we will be able to go back and refactor out the common stuff once we have the snippets set up.

File: src/interface/thirdparty-api.yaml (hunk 1 above)
Comment (eoln) [reply src/interface/thirdparty-api.yaml:46]: /health and /metrics are standard endpoints I've seen in mojaloop microservices

Comment (sridharvoruganti) [reply src/interface/thirdparty-api.yaml:46]: OK

**Pull request merged**